### PR TITLE
Custom alertmanager template example

### PR DESCRIPTION
**Pull request by Blizter**
This example answers #784 regarding the use of a custom alermanager template file that is decoupled from the alertmanager secret.

1. Create a `.tmpl file`, define your templates
2. reference path to templater in you alertmanager configuration
3. follow `alertmanager-config-template-external.jsonnet` to generate the template as a configmap then inject it inside the alermanager pod

before merging this PR, please merge #739 first, please :)

### CODE REVIEW

File: examples/alertmanager-config-template-external.jsonnet [hunk 1]
```diff
@@ -0,0 +1,28 @@
+local configmap(name, namespace, data) = {
+    apiVersion: "v1",
+    kind: "ConfigMap",
+    metadate : {
```
Comment (rohitdurvasula) [inline examples/alertmanager-config-template-external.jsonnet:4]: should be metadata

**Comment by metalmatze** [comment]
If you can fix the metadata typo I'm happy to have this as example going forward :blush:

**Comment by Blizter** [comment]
> If you can fix the metadata typo I'm happy to have this as example going forward

I'll do it, sorry for this taking time :)

**Comment by paulfantom** [comment]
@Blizter this example is still valid, we just need that one typo to be fixed.

**Review by PhilipGough — APPROVED** [review]
lgtm

**Pull request merged**